src/Microsoft.Azure.ServiceBus/Primitives/ServiceBusConnection.cs

```diff
@@ -51,7 +51,7 @@ public Task CloseAsync()
 
         protected void InitializeConnection(ServiceBusConnectionStringBuilder builder)
         {
-            this.Endpoint = builder.Endpoint;
+            this.Endpoint = new Uri(builder.Endpoint);
```

Comment:
@nemakam I made reference to the `ServiceBusConnection.NamespaceName` I've [added](https://github.com/Azure/azure-service-bus-dotnet/pull/193/files#diff-c938c0de58dbc4dacfdb4361f800aceaR106). While `Uri` does it all, it's not SBConnection's responsibility to determine the namespace name I'd think. Not a prio, but something I wanted to raise 🙂